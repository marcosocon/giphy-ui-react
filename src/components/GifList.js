import React from "react";
import { uniqueId, isEmpty, find } from 'lodash';

import { Container, Grid, Image, Button, Header } from 'semantic-ui-react';

function GifList({ title, favorites, gifsToShow, onAddFavorite, onRemoveFavorite }) {
    const showFavoriteHandler = (item) => {
        const isAlreadyAFavorite = !!find(favorites, (el) => el.id === item.id);
        return (
            <Button
                circular
                icon="heart"
                className="like"
                color={isAlreadyAFavorite ? 'black' : 'red'}
                onClick={() => isAlreadyAFavorite ? onRemoveFavorite(item) : onAddFavorite(item)}
            />
        )
    }

    return gifsToShow && !isEmpty(gifsToShow) ? (
        <Container>
            <Header
                as="h1"
                className="h1">
                {title}
            </Header>
            <Grid columns={4}>
                {gifsToShow.map((item, key) => {
                    return (
                        <Grid.Column key={uniqueId("trending-")}>
                            <Image
                                src={item.images.fixed_height.url}
                                as="a"
                                href={item.images.original.url}
                                target="_blank"
                                bordered
                            />
                            {showFavoriteHandler(item)}
                        </Grid.Column>
                    )
                })}
            </Grid>
        </Container>
    ) : null;
};

export default GifList;