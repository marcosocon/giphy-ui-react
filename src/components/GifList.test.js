import React from 'react';
import { mount, shallow } from 'enzyme';
import GifList from './GifList';

describe('<GifList />', () => {
    const gifsData = [{
        id: "MOCK1",
        images: {
            fixed_height: { url: "https://media2.giphy.com/media/zSUanrp1uZHfq/200.gif" },
            original: { url: "https://media2.giphy.com/media/zSUanrp1uZHfq/giphy.gif" },
        }
    }, {
        id: "MOCK2",
        images: {
            fixed_height: { url: "https://media2.giphy.com/media/zSUanrp1uZHfq/200.gif" },
            original: { url: "https://media2.giphy.com/media/zSUanrp1uZHfq/giphy.gif" },
        }
    }];
    it('render itself', () => {
        const component = shallow(<GifList gifsToShow={gifsData} favorites={gifsData} title='test' />);
        expect(component).toMatchSnapshot();
    });
    it('should show 2 items since we\'re passing two items in the gifs array', () => {
        const component = mount(<GifList gifsToShow={gifsData} favorites={gifsData} title='test' />);
        expect(component.find('.ui.container').length).toEqual(1);
        expect(component.find('.ui.header.h1').length).toEqual(1);
        expect(component.find('GridColumn').length).toEqual(2);
    });
});