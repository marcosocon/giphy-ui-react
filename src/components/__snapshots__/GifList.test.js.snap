// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GifList /> render itself 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <GifList
    favorites={
      Array [
        Object {
          "id": "MOCK1",
          "images": Object {
            "fixed_height": Object {
              "url": "https://media2.giphy.com/media/zSUanrp1uZHfq/200.gif",
            },
            "original": Object {
              "url": "https://media2.giphy.com/media/zSUanrp1uZHfq/giphy.gif",
            },
          },
        },
        Object {
          "id": "MOCK2",
          "images": Object {
            "fixed_height": Object {
              "url": "https://media2.giphy.com/media/zSUanrp1uZHfq/200.gif",
            },
            "original": Object {
              "url": "https://media2.giphy.com/media/zSUanrp1uZHfq/giphy.gif",
            },
          },
        },
      ]
    }
    gifsToShow={
      Array [
        Object {
          "id": "MOCK1",
          "images": Object {
            "fixed_height": Object {
              "url": "https://media2.giphy.com/media/zSUanrp1uZHfq/200.gif",
            },
            "original": Object {
              "url": "https://media2.giphy.com/media/zSUanrp1uZHfq/giphy.gif",
            },
          },
        },
        Object {
          "id": "MOCK2",
          "images": Object {
            "fixed_height": Object {
              "url": "https://media2.giphy.com/media/zSUanrp1uZHfq/200.gif",
            },
            "original": Object {
              "url": "https://media2.giphy.com/media/zSUanrp1uZHfq/giphy.gif",
            },
          },
        },
      ]
    }
    title="test"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <Header
          as="h1"
          className="h1"
        >
          test
        </Header>,
        <Grid
          columns={4}
        >
          <GridColumn>
            <Image
              as="a"
              bordered={true}
              href="https://media2.giphy.com/media/zSUanrp1uZHfq/giphy.gif"
              src="https://media2.giphy.com/media/zSUanrp1uZHfq/200.gif"
              target="_blank"
              ui={true}
            />
            <Button
              as="button"
              circular={true}
              className="like"
              color="black"
              icon="heart"
              onClick={[Function]}
            />
          </GridColumn>
          <GridColumn>
            <Image
              as="a"
              bordered={true}
              href="https://media2.giphy.com/media/zSUanrp1uZHfq/giphy.gif"
              src="https://media2.giphy.com/media/zSUanrp1uZHfq/200.gif"
              target="_blank"
              ui={true}
            />
            <Button
              as="button"
              circular={true}
              className="like"
              color="black"
              icon="heart"
              onClick={[Function]}
            />
          </GridColumn>
        </Grid>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "as": "h1",
          "children": "test",
          "className": "h1",
        },
        "ref": null,
        "rendered": "test",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <GridColumn>
              <Image
                as="a"
                bordered={true}
                href="https://media2.giphy.com/media/zSUanrp1uZHfq/giphy.gif"
                src="https://media2.giphy.com/media/zSUanrp1uZHfq/200.gif"
                target="_blank"
                ui={true}
              />
              <Button
                as="button"
                circular={true}
                className="like"
                color="black"
                icon="heart"
                onClick={[Function]}
              />
            </GridColumn>,
            <GridColumn>
              <Image
                as="a"
                bordered={true}
                href="https://media2.giphy.com/media/zSUanrp1uZHfq/giphy.gif"
                src="https://media2.giphy.com/media/zSUanrp1uZHfq/200.gif"
                target="_blank"
                ui={true}
              />
              <Button
                as="button"
                circular={true}
                className="like"
                color="black"
                icon="heart"
                onClick={[Function]}
              />
            </GridColumn>,
          ],
          "columns": 4,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "trending-1",
            "nodeType": "function",
            "props": Object {
              "children": Array [
                <Image
                  as="a"
                  bordered={true}
                  href="https://media2.giphy.com/media/zSUanrp1uZHfq/giphy.gif"
                  src="https://media2.giphy.com/media/zSUanrp1uZHfq/200.gif"
                  target="_blank"
                  ui={true}
                />,
                <Button
                  as="button"
                  circular={true}
                  className="like"
                  color="black"
                  icon="heart"
                  onClick={[Function]}
                />,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "as": "a",
                  "bordered": true,
                  "href": "https://media2.giphy.com/media/zSUanrp1uZHfq/giphy.gif",
                  "src": "https://media2.giphy.com/media/zSUanrp1uZHfq/200.gif",
                  "target": "_blank",
                  "ui": true,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "as": "button",
                  "circular": true,
                  "className": "like",
                  "color": "black",
                  "icon": "heart",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "trending-2",
            "nodeType": "function",
            "props": Object {
              "children": Array [
                <Image
                  as="a"
                  bordered={true}
                  href="https://media2.giphy.com/media/zSUanrp1uZHfq/giphy.gif"
                  src="https://media2.giphy.com/media/zSUanrp1uZHfq/200.gif"
                  target="_blank"
                  ui={true}
                />,
                <Button
                  as="button"
                  circular={true}
                  className="like"
                  color="black"
                  icon="heart"
                  onClick={[Function]}
                />,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "as": "a",
                  "bordered": true,
                  "href": "https://media2.giphy.com/media/zSUanrp1uZHfq/giphy.gif",
                  "src": "https://media2.giphy.com/media/zSUanrp1uZHfq/200.gif",
                  "target": "_blank",
                  "ui": true,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "as": "button",
                  "circular": true,
                  "className": "like",
                  "color": "black",
                  "icon": "heart",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <Header
            as="h1"
            className="h1"
          >
            test
          </Header>,
          <Grid
            columns={4}
          >
            <GridColumn>
              <Image
                as="a"
                bordered={true}
                href="https://media2.giphy.com/media/zSUanrp1uZHfq/giphy.gif"
                src="https://media2.giphy.com/media/zSUanrp1uZHfq/200.gif"
                target="_blank"
                ui={true}
              />
              <Button
                as="button"
                circular={true}
                className="like"
                color="black"
                icon="heart"
                onClick={[Function]}
              />
            </GridColumn>
            <GridColumn>
              <Image
                as="a"
                bordered={true}
                href="https://media2.giphy.com/media/zSUanrp1uZHfq/giphy.gif"
                src="https://media2.giphy.com/media/zSUanrp1uZHfq/200.gif"
                target="_blank"
                ui={true}
              />
              <Button
                as="button"
                circular={true}
                className="like"
                color="black"
                icon="heart"
                onClick={[Function]}
              />
            </GridColumn>
          </Grid>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "as": "h1",
            "children": "test",
            "className": "h1",
          },
          "ref": null,
          "rendered": "test",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <GridColumn>
                <Image
                  as="a"
                  bordered={true}
                  href="https://media2.giphy.com/media/zSUanrp1uZHfq/giphy.gif"
                  src="https://media2.giphy.com/media/zSUanrp1uZHfq/200.gif"
                  target="_blank"
                  ui={true}
                />
                <Button
                  as="button"
                  circular={true}
                  className="like"
                  color="black"
                  icon="heart"
                  onClick={[Function]}
                />
              </GridColumn>,
              <GridColumn>
                <Image
                  as="a"
                  bordered={true}
                  href="https://media2.giphy.com/media/zSUanrp1uZHfq/giphy.gif"
                  src="https://media2.giphy.com/media/zSUanrp1uZHfq/200.gif"
                  target="_blank"
                  ui={true}
                />
                <Button
                  as="button"
                  circular={true}
                  className="like"
                  color="black"
                  icon="heart"
                  onClick={[Function]}
                />
              </GridColumn>,
            ],
            "columns": 4,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "trending-1",
              "nodeType": "function",
              "props": Object {
                "children": Array [
                  <Image
                    as="a"
                    bordered={true}
                    href="https://media2.giphy.com/media/zSUanrp1uZHfq/giphy.gif"
                    src="https://media2.giphy.com/media/zSUanrp1uZHfq/200.gif"
                    target="_blank"
                    ui={true}
                  />,
                  <Button
                    as="button"
                    circular={true}
                    className="like"
                    color="black"
                    icon="heart"
                    onClick={[Function]}
                  />,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "as": "a",
                    "bordered": true,
                    "href": "https://media2.giphy.com/media/zSUanrp1uZHfq/giphy.gif",
                    "src": "https://media2.giphy.com/media/zSUanrp1uZHfq/200.gif",
                    "target": "_blank",
                    "ui": true,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "as": "button",
                    "circular": true,
                    "className": "like",
                    "color": "black",
                    "icon": "heart",
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "trending-2",
              "nodeType": "function",
              "props": Object {
                "children": Array [
                  <Image
                    as="a"
                    bordered={true}
                    href="https://media2.giphy.com/media/zSUanrp1uZHfq/giphy.gif"
                    src="https://media2.giphy.com/media/zSUanrp1uZHfq/200.gif"
                    target="_blank"
                    ui={true}
                  />,
                  <Button
                    as="button"
                    circular={true}
                    className="like"
                    color="black"
                    icon="heart"
                    onClick={[Function]}
                  />,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "as": "a",
                    "bordered": true,
                    "href": "https://media2.giphy.com/media/zSUanrp1uZHfq/giphy.gif",
                    "src": "https://media2.giphy.com/media/zSUanrp1uZHfq/200.gif",
                    "target": "_blank",
                    "ui": true,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "as": "button",
                    "circular": true,
                    "className": "like",
                    "color": "black",
                    "icon": "heart",
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
